### DISCUSSION TITLE
Wierd window stretch when using a tiling wm (linux, wayland, gtk)

### DISCUSSION THREAD

**Discussion by M4rti21** [ghostty-org/ghostty#3592] [Q&A]
Whenever i create new windows (which makes the existing ones resize) i get this weird stretches.

https://github.com/user-attachments/assets/c993c0d8-eb31-4da2-baba-ea06fd29ae1e

- Build: `ghostty 1.0.0-3` from the Arch Linux repository.
- OS: Artix Linux
- WM: dwl (wayland wlroots)
- Backend: pretty sure its libadwaita...

My config:
```conf
foreground = "C5C9C5"
background = "12120f"
background-opacity = 0.95
background-blur-radius = 15

theme = "Kanagawa Dragon"

font-family="IBM Plex Mono Medium"
font-family-bold="IBM Plex Mono Bold"
font-family-italic="IBM Plex Mono Medium Italic"
font-family-bold-italic="IBM Plex Mono Bold Italic"

window-decoration = false
window-save-state = "never"
window-padding-balance = true
window-padding-y = 8
window-padding-x = 8

resize-overlay = "never"

clipboard-read = allow
clipboard-write = allow
clipboard-paste-protection = false

confirm-close-surface = false
quit-after-last-window-closed = true

gtk-single-instance = true
gtk-titlebar = false

bold-is-bright = true
```

**Comment by tristan957** [comment]
This seems related to your window manager. Can you check on sway or something a little more popular?

**Reply by M4rti21** [reply]
the same happens when im on Hyprland, haven't tried sway, but given that dwl is also based on wlroots, and hyprland uses a fork of it, i imagine i'd get the same result :(

**Reply by tristan957** [reply]
I'd recommend finding some other dwl or hyprland users on Discord just to see if they have the same issue.

**Comment by cosarara** [comment]
I can reproduce the same issue on X11. Just before the terminal is properly resized, I see a couple frames of stretched text.

* Build: Ghostty 1.0.1-dev+0000000 (from the archlinux repo)
* OS: Arch linux
* WM: awesomewm (no compositor running)
* Backend: gtk4, libadwaita, X11
* gpu: Intel UHD Graphics 620
* Empty ghostty config

In the video I close and open a terminal on the left, which resizes the one on the right:
https://github.com/user-attachments/assets/4679f5d7-645a-4e70-b02c-401de4e0b41b

Going from windowed to fullscreen on a single terminal also shows the same stretching.

**Reply by tristan957** [reply]
It seems like Ghostty is slow to repaint. I'm not sure all the mechanics that would go into this. It would require reading some code or someone smarting.

**Reply by M4rti21** [reply]
ok i just found out if i explicitly enable libadwaita via `gtk-adwaita = true` the error doesnt occur, if i then toggle it off with `gtk-adwaita = false` it breaks as shown in the post...

**Reply by tristan957** [reply]
Is the the only way you can reproduce it, only toggling the option?

**Reply by M4rti21** [reply]
that seems to be the case

**Reply by cosarara** [reply]
I have the issue both with and without gtk-adwaita=true. The workaround does not seem to do anything on my machine.

**Reply by deifactor** [reply]
Also replicates for me:

```
Ghostty 1.1.3

Version
  - version: 1.1.3
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - desktop env: other
  - GTK version:
    build      : 4.18.5
    runtime    : 4.18.5
  - libadwaita : enabled
    build      : 1.7.3
    runtime    : 1.7.3
  - libX11     : enabled
  - libwayland : enabled
```

Running under sway on NixOS. `gtk-adwaita` has no effect.
